### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by avneetmalhotra**

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -1,5 +1,18 @@
 class AdminController < ApplicationController
+  before_action :ensure_admin
+
   def index
+    redirect_to admin_reports_url if current_user.role == 'admin'
```
Comment (nimish13) [inline app/controllers/admin_controller.rb]: now /admin route is not accessible as we are redirecting in both cases
create a Admin::BaseController and inherit all controller in admin namespace from it

File: app/controllers/admin/reports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Admin::ReportsController < AdminController  
+
+  def index
+    if params[:from_date]
```
Comment (nimish13) [inline app/controllers/admin/reports_controller.rb]: do this in a before_action

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -8,6 +9,7 @@ def create
     user = User.find_by(name: params[:name])
     if user.try(:authenticate, params[:password])
       session[:user_id] = user.id
+      session[:last_activity_time] = Time.now
```
Comment (nimish13) [inline app/controllers/sessions_controller.rb]: use Time.current in entire application

**Comment by avneetmalhotra** [comment]
closed in reference to #11

**Pull request closed without merging**